lib/ts/Extendible.h

```diff
@@ -0,0 +1,584 @@
+/** @file
+
+  Extendible
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  @section details Details
+
+////////////////////////////////////////////
+  Implements Extendible<Derived_t>:
+  * Extendible<Derived_t>::Schema
+  * Extendible<Derived_t>::FieldSchema
+  * Extendible<Derived_t>::FieldId<Access_t,Field_t>
+ */
+
+#pragma once
+#include "stdint.h"
+#include <typeinfo>
+#include <typeindex>
+#include <cstddef>
+#include <atomic>
+#include <cstring>
+#include <unordered_map>
+#include <type_traits>
+
+#include "ts/ink_memory.h"
+#include "ts/ink_assert.h"
+#include "AcidPtr.h"
+
+#ifndef ROUNDUP
+#define ROUNDUP(x, y) ((((x) + ((y)-1)) / (y)) * (y))
+#endif
+
+// used for C API
+typedef const void *FieldId_C;
+
+namespace MT // (multi-threaded)
+{
+/// used to store byte offsets to fields
+using ExtendibleOffset_t = uint16_t;
+/// all types must allow unblocking MT read access
+enum AccessEnum { ATOMIC, BIT, STATIC, ACIDPTR, DIRECT, C_API, NUM_ACCESS_TYPES };
+
+inline bool &
+areStaticsFrozen()
+{
+  static bool frozen = 0;
```

Comment:
How is this ever set to non-zero? Are you depending on the raw memory not being zero before static initialization?